index.js

```diff
@@ -313,6 +313,65 @@ MochaJUnitReporter.prototype.getTestsuiteData = function(suite) {
   return testSuite;
 };
 
+/**
+ * Given a fully qualified test name and test case context, derive a list of user-defined
+ * tags, if present.
+ *
+ * @param {String} name - the fully qualified name of the test
+ * @param {object} test - test case
+ */
+function getTagsFromTestcaseData(name, test) {
+  // extract tags and teams from the test name
+  // this regex matches any whole word beginning with # or @
+  // includes words surrounded by parenthesis, square brackets, and punctuation
+  var tagsFromName = name.match(/([#|@]\w+)/g) || [];
+
+  // inspect the supplied test object to see if it has any user-created properties
+  // this object will be present if the tests were run by Cypress, and if any additional
+  // key/value pairs were added to the test.
+  var testConfig = test && test._testConfig || {};
+
+  var tagPropertyValue = undefined;
+
+  // Cypress => ~9.0.0 puts user-defined tags in _testConfig.unverifiedTestConfig.tags
+  if (testConfig.unverifiedTestConfig && testConfig.unverifiedTestConfig.tags) {
+    tagPropertyValue = testConfig.unverifiedTestConfig.tags;
+  }
+  // Cypress =< ~8.0.0 puts user-defined tags in _testConfig.tags
+  else if (testConfig.tags) {
+    tagPropertyValue = testConfig.tags;
+  }
+
+  var tagsFromProperties = [];
+
+  // the 'tags' property can contain tags in a comma separated string, or in an array
+  // figure out which format is used, and ensure we end up with an array
+  if (tagPropertyValue) {
+    if (Array.isArray(tagPropertyValue)) {
+      tagsFromProperties = tagPropertyValue;
+    } else if (typeof tagPropertyValue === 'string' || tagPropertyValue instanceof String) {
```

Comment:
throwing these out there in case we want an extra robust check for whether or not it's a string

https://lodash.com/docs/4.17.15#isString or their source: https://github.com/lodash/lodash/blob/2da024c3b4f9947a48517639de7560457cd4ec6c/isString.js#L20

Reply:
i wrestled with the idea of adding lodash but didn't want to introduce a dependency and figured i'd wing it.  i hadn't considered just copying the lodash functions.

lodash is MIT licensed so we can do so without issue; i've added `isString` and the dependency `getTag`, and created `LICENSE-THIRD-PARTY.txt` in the root so we're attributing lodash properly.